lib/tbot/service_bot_identity.go

```diff
@@ -470,9 +488,31 @@ func renewIdentity(
 		return newIdentity, nil
 	}
 
+	now := time.Now()
+	if expiry, ok := facade.Expiry(); !ok || now.After(expiry) {
```

Comment:
My only thought with client-side checks of a certificates validity is that the certificate /could/ be valid but the client could be heavily drifted, but, I think in that case we'd probably want to encourage people to fix the drift rather than adjusting the product to accomodate it.